cpuprofiler.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Roger Chapman and the v8go contributors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package v8go
+
+// #include <stdlib.h>
+// #include "v8go.h"
+import "C"
+import (
+	"unsafe"
+)
+
+type CPUProfiler struct {
+	ptr C.CpuProfilerPtr
+	iso *Isolate
+}
+
+// CPUProfiler is used to control CPU profiling.
+func NewCPUProfiler(iso *Isolate) *CPUProfiler {
+	return &CPUProfiler{
+		ptr: C.NewCpuProfiler(iso.ptr),
+		iso: iso,
+	}
+}
+
+// Dispose will dispose the profiler; subsequent calls will panic.
+func (c *CPUProfiler) Dispose() {
+	if c.ptr == nil {
+		return
+	}
+	C.CpuProfilerDispose(c.ptr)
+	c.ptr = nil
+}
+
+// StartProfiling starts collecting a CPU profile. Title may be an empty string. Several
+// profiles may be collected at once. Attempts to start collecting several
+// profiles with the same title are silently ignored.
+func (c *CPUProfiler) StartProfiling(title string) {
+	tstr := C.CString(title)
+	defer C.free(unsafe.Pointer(tstr))
+	C.CpuProfilerStartProfiling(c.iso.ptr, c.ptr, tstr)
```

Comment:
Documentation for CPUProfiler.Dispose says "subsequent calls will panic", but there are no corresponding checks for `c.iso.ptr` or `c.ptr` being `nil` to panic if either has been disposed.

Reply:
I actually don't know where I got that documentation/comment from. I originally was basing myself on the v8 api docs online then source directly from the header files but neither that I'm looking at right now suggest that this will in fact occur. The `v8-profiler.h` describes `CPUProfiler.Dispose` as `   * Disposes the CPU profiler object.` I'll update the comment to drop that bit unless we prefer the behavior to be a panic in these situations at the go level than when it hits the c++ implementation.

Reply:
Oh I likely examined the behavior on an isolate: https://github.com/rogchap/v8go/blob/10e762dc4017337a7033a09ba3d2a0efa866ce05/isolate.go#L97

In this case, I'm not sure the comment is right either. Is it suggesting that subsequent calls to `Dispose` will panic or subsequent calls on any functions of the isolate will? The former isn't what's actually happening and there's a test to verify that: https://github.com/rogchap/v8go/blob/10e762dc4017337a7033a09ba3d2a0efa866ce05/isolate_test.go#L110